### ISSUE TITLE
method `xxx` is not a member of trait `XXX` doesn't check spelling

### ISSUE THREAD

**Issue by guswynn** [rust-lang/rust#85942]
Today I hit this issue: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=fdce0c73a1c1190a8bd4a82c5cc24b52 

but I couldn't for the life of me see the difference in the methods.

In some places, rust suggests differences that are short Levenshtein distance from each other, can that be done here as well?

**Comment by hellow554** [comment]
It's done for structs and their fields/methods, but not for traits, e.g.:

```rust
struct A;
impl A {
    fn away(self) {}
}

fn main() {
    A.aay();
}
```

```
error[E0599]: no method named `aay` found for struct `A` in the current scope
  --> src/main.rs:16:7
   |
10 | struct A;
   | --------- method `aay` not found for this
...
16 |     A.aay();
   |       ^^^ help: there is an associated function with a similar name: `away`
```


If you would implement this for traits, there's a good chance that you might get a lof of false positives, but I can see your point, so I think this should be implemented! :)

**Comment by guswynn** [comment]
@hellow554 for calling trait methods I agree, we should be careful about false-positives, but this issue is more specifically about impl-blocks for traits, where we know the exact list of required methods to implement to compare to

**Comment by hellow554** [comment]
Ah! Yes! You're absolutely right! That's a great thing to do :D
